src/mainwindow.cpp

```diff
@@ -542,7 +545,70 @@ void MainWindow::setupActionIcon()
     ui->actionViewLog->setIcon(QIcon::fromTheme("view-list-text",
                                QIcon::fromTheme("text-x-preview")));
     ui->actionGeneralSettings->setIcon(QIcon::fromTheme("configure",
-                                   QIcon::fromTheme("preferences-desktop")));
+                                       QIcon::fromTheme("preferences-desktop")));
     ui->actionReportBug->setIcon(QIcon::fromTheme("tools-report-bug",
                                  QIcon::fromTheme("help-faq")));
 }
+
+bool MainWindow::isInstanceRunning() const
+{
+    return instanceRunning;
+}
+
+void MainWindow::initSingleInstance()
+{
+    instanceRunning = false;
+
+    QString username = qgetenv("USER");
+    if (username.isEmpty()){
+        username = qgetenv("USERNAME");
+    }
+
+    QString serverName = QCoreApplication::applicationName() + "_" + username;
+    QLocalSocket socket;
+    socket.connectToServer(serverName);
+    if(socket.waitForConnected(500)){
```

Comment:
Space after `if` and space before `{` would be appreciated. Same applies for elsewhere in this request.